### PR TITLE
Translate DemonsRegistration1-2 and DicomSeriesFromArray to R

### PR DESCRIPTION
**Pull request by seungkyoon**
Translate DemonsRegistration1 and DemonsRegistration2 Python examples
to R, add R tests to CMakeLists.txt, and modify Documentation.rst to reflect
new examples

### CODE REVIEW

File: Examples/DemonsRegistration1/DemonsRegistration1.R [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+matcher <- HistogramMatchingImageFilter()
+matcher$SetNumberOfHistogramLevels(1024)
+matcher$SetNumberOfMatchPoints(7)
+matcher$ThresholdAtMeanIntensityOn()
+moving <- matcher$Execute(moving, fixed)
+
+demons <- DemonsRegistrationFilter()
+demons$SetNumberOfIterations( 50 )
+demons$SetStandardDeviations( 1.0 )
```
Comment (zivy) [inline Examples/DemonsRegistration1/DemonsRegistration1.R]: What does the standard deviation control?

File: Examples/DemonsRegistration1/DemonsRegistration1.R [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+moving <- ReadImage(args[[2]], 'sitkFloat32')
+
+
+matcher <- HistogramMatchingImageFilter()
+matcher$SetNumberOfHistogramLevels(1024)
+matcher$SetNumberOfMatchPoints(7)
+matcher$ThresholdAtMeanIntensityOn()
+moving <- matcher$Execute(moving, fixed)
+
+demons <- DemonsRegistrationFilter()
```
Comment (zivy) [inline Examples/DemonsRegistration1/DemonsRegistration1.R]: State that this is the "basic" demons algorithm and that there is a family of such algorithms in SimpleITK.

File: Examples/DemonsRegistration2/DemonsRegistration2.R [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#  limitations under the License.
+#
+#=========================================================================
+# Run with:
+#
+# Rscript --vanilla DemonsRegistration2.R fixedImageFilter movingImageFile [initialTransformFile] outputTransformFile
+#
+
+library(SimpleITK)
+
```
Comment (zivy) [inline Examples/DemonsRegistration2/DemonsRegistration2.R]: Please see the comments on the Demons1 example, all apply here (only difference is the use of a fast symmetric forces variant).

**Comment by seungkyoon** [comment]
Made changes and added DicomSeriesFromArray R example

**Comment by blowekamp** [comment]
topic merged to master

**Pull request merged**